channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -545,8 +505,71 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
+                        _self._appId = response.appId;
+                    }
+                }
+    
+                if ((status < 200 || status >= 300) && status !== 0) {
+
+                    // Update End Point url if permanent redirect or moved permanently
+                    // Updates the end point url before retry
+                    if(status === 301 || status === 308 ) {
+                        if(!_checkAndUpdateEndPointUrl(responseUrl)) {
+                            _self._onError(payload, errorMessage);
+                            return;
+                        }
+                    }
+
+                    if (!_self._senderConfig.isRetryDisabled() && _isRetriable(status)) {
+                        _resendPayload(payload);
+                        _self.diagLog().throwInternal(
+                            LoggingSeverity.WARNING,
+                            _InternalMessageId.TransmissionFailed, ". " +
+                            "Response code " + status + ". Will retry to send " + payload.length + " items.");
+                    } else {
+                        _self._onError(payload, errorMessage);
+                    }
+                } else if (Offline.isOffline()) { // offline
+                    // Note: Don't check for status == 0, since adblock gives this code
+                    if (!_self._senderConfig.isRetryDisabled()) {
+                        const offlineBackOffMultiplier = 10; // arbritrary number
+                        _resendPayload(payload, offlineBackOffMultiplier);
+    
+                        _self.diagLog().throwInternal(
+                            LoggingSeverity.WARNING,
+                            _InternalMessageId.TransmissionFailed, `. Offline - Response Code: ${status}. Offline status: ${Offline.isOffline()}. Will retry to send ${payload.length} items.`);
+                    }
+                } else {
+
+                    // check if the xhr's responseURL is same as endpoint url
+                    // TODO after 10 redirects force send telemetry with 'redirect=false' as query parameter.
+                    _checkAndUpdateEndPointUrl(responseUrl);
+                    
+                    if (status === 206) {
+                        if (!response) {
+                            response = _parseResponse(res);
+                        }
+    
+                        if (response && !_self._senderConfig.isRetryDisabled()) {
+                            _self._onPartialSuccess(payload, response);
+                        } else {
+                            _self._onError(payload, errorMessage);
+                        }
+                    } else {
+                        _consecutiveErrors = 0;
+                        _self._onSuccess(payload, countOfItemsInPayload);
+                    }
+                }
+            }
+
+            function _checkAndUpdateEndPointUrl(responseURL: string) {
+                const responseUrl = responseURL;
```

Comment:
Why the local assignment? As your passing the string value now -- just rename the param.